### PR TITLE
mysql_jdbc: modernize

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/sql/mysql/jdbc/default.nix [hunk 1]
```diff
@@ -1,18 +1,32 @@
...
 
 stdenv.mkDerivation rec {
-  name = "mysql-connector-java-5.1.46";
-  builder = ./builder.sh;
+  pname = "mysql-connector-java";
+  version = "5.1.46";
 
   src = fetchurl {
-    url = "http://dev.mysql.com/get/Downloads/Connector-J/${name}.zip";
+    url = "http://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-java-${version}.zip";
```
Comment (prusnak) [inline pkgs/servers/sql/mysql/jdbc/default.nix:8]: Let's use https and pname

```suggestion
    url = "https://dev.mysql.com/get/Downloads/Connector-J/${pname}-${version}.zip";
```

File: pkgs/servers/sql/mysql/jdbc/default.nix [hunk 2]
```diff
@@ -1,18 +1,32 @@
...
     sha256 = "0dfjshrrx0ndfb6xbdpwhn1f1jkw0km57rgpar0ny8ixmgdnlwnm";
   };
 
+  unpackPhase = ''
```
Comment (prusnak) [inline pkgs/servers/sql/mysql/jdbc/default.nix:12]: Doesn't fetchurl unzip automatically?

File: pkgs/servers/sql/mysql/jdbc/default.nix [hunk 3]
```diff
@@ -1,18 +1,32 @@
...
+  '';
+
   nativeBuildInputs = [ unzip ];
+
   buildInputs = [ ant ];
 
-  meta = {
-    platforms = lib.platforms.unix;
+  meta = with lib; {
+    description = " MySQL Connector/J ";
```
Comment (prusnak) [inline pkgs/servers/sql/mysql/jdbc/default.nix:26]: Trim spaces:

```suggestion
    description = "MySQL Connector/J";
```

File: pkgs/servers/sql/mysql/jdbc/default.nix [hunk 4]
```diff
@@ -1,18 +1,32 @@
-{lib, stdenv, fetchurl, ant, unzip}:
+{ lib, stdenv, fetchurl, ant, unzip }:
 
 stdenv.mkDerivation rec {
-  name = "mysql-connector-java-5.1.46";
-  builder = ./builder.sh;
+  pname = "mysql-connector-java";
+  version = "5.1.46";
```
Comment (prusnak) [inline pkgs/servers/sql/mysql/jdbc/default.nix:5]: Latest of 5.x series is `5.1.49`

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 131529` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>atlassian-confluence</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>jboss_mysql_jdbc</li>
    <li>keycloak</li>
    <li>liquibase</li>
    <li>mysql_jdbc</li>
    <li>squirrel-sql</li>
    <li>tomcat_mysql_jdbc</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131529` at 3c188357 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built successfully:</summary>
  <ul>
    <li>jboss_mysql_jdbc</li>
    <li>keycloak</li>
    <li>liquibase</li>
    <li>mysql_jdbc</li>
    <li>squirrel-sql</li>
    <li>tomcat_mysql_jdbc</li>
  </ul>
</details>

**Pull request merged**